internal/worker/dbaccessor/tracker.go

```diff
@@ -380,10 +389,10 @@ func (w *trackedDBWorker) loop() error {
 // ensureDBAliveAndOpenNewIfRequired is a bit long-winded, but it is a way to
 // ensure that the underlying database is alive and well. If it is not, we
 // attempt to open a new one. If that fails, we return an error.
-func (w *trackedDBWorker) ensureDBAliveAndOpenIfRequired(db *sql.DB) (*sql.DB, error) {
+func (w *trackedDBWorker) ensureDBAliveAndOpenIfRequired(ctx context.Context, db *sql.DB) (*sql.DB, error) {
 	// Allow killing the tomb to cancel the context,
 	// so shutdown/restart can not be blocked by this call.
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
 	ctx = w.tomb.Context(ctx)
```

Comment:
**todo:** This line should be removed as well (bounding context to tomb is now handled outside the function l343:

https://github.com/juju/juju/pull/18231/files#diff-4dcf535ef6cc5d3f34ed25bc3d44275803733410d9e05274130ac53d89ce206fR343

Reply:
Cool, deleted that wrapper